src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+	return performWriteQuery([ {
+		$: `UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000,
+			lastupdatetime = &{lastupdatetime}, mtbu = &{mtbu} 
+			WHERE url=&{url}`,
+		lastupdatetime,
+		url,
+		mtbu
+	} ]);
+}
+
+function updateFeedsOnNoNewArtcles (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: 'UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000 WHERE url=&{url}',
+		url
+	} ]);
+}
+
+export const newsAggregator = (): Promise<Array<any>> => {
+	// Check if the feeds table is not empty.
+	return performReadQuery({
+		$: 'SELECT url FROM feeds',
+	})
+	.then((results) => {
+		// results: [{url: _}, ...]
+		if (results.length === 0) {
+			winston.info('feeds table is empty');
+			// insert seed url if feeds table is empty
+			const seedUrl = 'http://bangalore.citizenmatters.in/rss';
+			return insertIntoFeeds(seedUrl);
+		}
+		return [ { rowCount: 0 } ];
+	})
+	.then(async (response) => {
+		// response: [{.., rowCount: _, ...}, ..]
+		if (response[0].rowCount === 1) {
+			winston.info('Insert operation performed successfully.');
+		}
+		await truncateArticlesTable();
+		// feeds: [{url: _, mtbu: _, lastrequesttime: _, lastupdatetime: _}, .....]
+		const feeds = await filterFeeds();
+		if (feeds.length > 0) {
+			const parseFeedPromises = [];
+			for (const feed of feeds) {
+				/*
+					CHECK FOR NEW ARTICLES:
+						- checks:
+							-> shouldn't be older than 1 week.
+							-> Avoid duplicate stroies from the same feed.
+							-> To avoid duplicate stories, compare article's pubdate with the lastupdatetime for that feed.
+				*/
+				const newArticles = [];
+				let latestArticleDate = feed.lastupdatetime;
+				parseFeedPromises.push(
+					parseNewsFeed(feed.url)
+					.then((newsArticles) => {
+						const promises = [];
+						for (const newsArticle of newsArticles) {
+							const articlePubDate = (new Date(newsArticle.published)).getTime();
+							const sevenDays = 7 * 24 * 60 * 60 * 1000;
+							if (hasNewNewsArticle(articlePubDate, feed.lastupdatetime) && articleAge(articlePubDate) <= sevenDays) {
+								latestArticleDate = Math.max(articlePubDate, latestArticleDate);
+								const article = _.pick(newsArticle, [ 'title', 'link', 'content', 'published' ]);
+								newArticles.push(article);
+							}
+						}
+
+						if (newArticles.length > 0) {
+							const mtbu = calculateMTBU(newArticles);
+							// update MTBU, lastrequesttime, lastupdatetime
+							promises.push(
+								updateFeedsOnNewArticles(latestArticleDate, feed.url, mtbu)
+							);
+
+							// Insert new articles
+							for (const article of newArticles) {
+								const url = article.link;
+								const articleObj = article;
+								const contentForLexemes = article.title + ' ' + article.content;
+								promises.push(
+									insertNewArticles(url, articleObj, contentForLexemes)
```

Comment:
Ouch. If a feed returns 20 different articles, you make 20 INSERT queries here. Do one INSERT with 20 rows for significant reduction in DB load.